scripts/download_checkpoints.py

```diff
@@ -0,0 +1,139 @@
+import csv
+import os
+import requests
+from tqdm import tqdm
+import argparse
+from pathlib import Path
+from urllib.parse import urljoin
+
+def convert_to_r2_url(http_url):
+    """Convert HTTP URL to R2 URL format."""
+    if http_url.startswith('https://olmo-checkpoints.org/'):
+        return http_url.replace('https://olmo-checkpoints.org/', 'r2://olmo-checkpoints/')
+    return http_url
+
+def convert_to_public_url(r2_url):
+    """Convert R2 URL to public HTTP URL format."""
+    if r2_url.startswith('r2://olmo-checkpoints/'):
+        return r2_url.replace('r2://olmo-checkpoints/', 'https://olmo-checkpoints.org/')
+    return r2_url
+
+def download_file(url, save_path, chunk_size=8192):
+    """Download a file with progress bar."""
+    response = requests.get(url, stream=True)
+    response.raise_for_status()
+    total_size = int(response.headers.get('content-length', 0))
+    save_path.parent.mkdir(parents=True, exist_ok=True)
+    
+    with open(save_path, 'wb') as f:
+        with tqdm(total=total_size, unit='B', unit_scale=True, desc=save_path.name) as pbar:
+            for chunk in response.iter_content(chunk_size=chunk_size):
+                if chunk:
+                    f.write(chunk)
+                    pbar.update(len(chunk))
+
+def try_get_directory_listing(url):
+    common_files = [
+        "config.yaml",
+        "model.pt",
+        "optim.pt",
+        "train.pt",
+        "model.safetensors",
+        "optim.safetensors",
+    ]
+    
+    found_files = []
+    for pattern in common_files:
+        test_url = urljoin(url.rstrip('/') + '/', pattern)
+        try:
+            response = requests.head(test_url)
+            if response.status_code == 200:
+                found_files.append(pattern)
+        except requests.exceptions.RequestException:
+            continue
```

Comment:
Why would you swallow these exceptions?

Reply:
Oh, because you don't expect all files to be there? Then at least catch only 404 errors.
But better yet, list the contents of the directory in one call to check what's there, instead of making six calls every time we have to check a directory.

Reply:
The website does not support directory listing, so I had to manually add these ("config.yaml", "model.pt", "optim.pt", "train.pt", "model.safetensors", "optim.safetensors")

Reply:
Ok, manually looking for them works too.